File: src/main/java/com/elf/vipForElf/domain/test/vo/AreaInfoVO.java
package com.elf.vipForElf.domain.test.vo;

import com.elf.vipForElf.domain.test.entity.AreaEntity;
import com.elf.vipForElf.domain.test.entity.FloorEntity;
import com.elf.vipForElf.web.dto.MoveInAreaDTO;
import com.elf.vipForElf.web.dto.NewAreaDTO;
import lombok.Value;

import java.time.OffsetDateTime;

@Value
public class AreaInfoVO {
    Long id;
    String status;
    String officeName;
    String roomNumber;
    String areaSize;
    OffsetDateTime moveInDate;
    FloorEntity floorEntity;

    public AreaInfoVO(NewAreaDTO newAreaDTO, FloorEntity floorEntity){
        this.id = null;
        this.status = null;
        this.officeName = null;
        this.roomNumber = newAreaDTO.getRoomNumber();
        this.moveInDate = null;
        this.areaSize = newAreaDTO.getAreaSize();
        this.floorEntity = floorEntity;
    }
    public AreaInfoVO(AreaEntity areaEntity){
        this.id = areaEntity.getId();
        this.status = areaEntity.getStatus();
        this.officeName = areaEntity.getOfficeName();
        this.roomNumber = areaEntity.getRoomNumber();
        this.moveInDate = areaEntity.getMoveInDate();
        this.areaSize = areaEntity.getAreaSize();
        this.floorEntity = areaEntity.getFloor();
    }
    public AreaInfoVO(MoveInAreaDTO moveInAreaDTO){
        this.id = moveInAreaDTO.getId();
        this.officeName = moveInAreaDTO.getOfficeName();
        this.status = null;
        this.areaSize = null;
        this.roomNumber = null;
        this.floorEntity = null;
        this.moveInDate = OffsetDateTime.now();
    }
}

File: src/main/java/com/elf/vipForElf/domain/area/vo/AreaInfoVO.kt
package com.elf.vipForElf.domain.area.vo

import com.elf.vipForElf.domain.area.entity.AreaEntity
import com.elf.vipForElf.domain.floor.entity.FloorEntity
import com.elf.vipForElf.web.dto.MoveInAreaDTO
import com.elf.vipForElf.web.dto.NewAreaDTO
import java.time.OffsetDateTime

data class AreaInfoVO(
    val id: Long? = null,
    val status: String? = null,
    val officeName: String? = null,
    val roomNumber: String? = null,
    val areaSize: String? = null,
    val moveInDate: OffsetDateTime? = null,
    val floorEntity: FloorEntity? = null
) {
    constructor(newAreaDTO: NewAreaDTO, floorEntity: FloorEntity) : this(
        id = null,
        status = null,
        officeName = null,
        roomNumber = newAreaDTO.roomNumber,
        moveInDate = null,
        areaSize = newAreaDTO.areaSize,
        floorEntity = floorEntity
    )

    constructor(areaEntity: AreaEntity) : this(
        id = areaEntity.id,
        status = areaEntity.status,
        officeName = areaEntity.officeName,
        roomNumber = areaEntity.roomNumber,
        moveInDate = areaEntity.moveInDate,
        areaSize = areaEntity.areaSize,
        floorEntity = areaEntity.floor
    )

    constructor(moveInAreaDTO: MoveInAreaDTO) : this(
        id = moveInAreaDTO.id,
        officeName = moveInAreaDTO.officeName,
        status = null,
        areaSize = null,
        roomNumber = null,
        floorEntity = null,
        moveInDate = OffsetDateTime.now()
    )
}
